packages/remix-react/data.ts

```diff
@@ -29,7 +29,8 @@ export function isErrorResponse(response: any): boolean {
 export function isRedirectResponse(response: any): boolean {
   return (
     response instanceof Response &&
-    response.headers.get("X-Remix-Redirect") != null
+    (response.headers.get("X-Remix-Redirect") != null ||
+      (response.status >= 300 && response.status < 400))
```

Comment:
We can steal this util from RR (or even export it as `UNSAFE_` and use it directly:

```ts
const redirectStatusCodes = new Set([301, 302, 303, 307, 308]);

export function isRedirectStatusCode(statusCode: number): boolean {
  return redirectStatusCodes.has(statusCode);
}
```

Reply:
Is the reason we need this because clientLoader doesn't run through the internal checking?  Maybe we can get rid of the need for this an put _less_ in the `doFetch`...

Reply:
The reason we need this is because Remix is expecting a 204 response to represent redirects and we don't currently handle *real* redirect responses.